### PR TITLE
prometheusremotewrite: Don't generate target_info unless there are metrics

### PR DESCRIPTION
**Pull request by aknuds1**
**Description:** <Describe what has changed.>
Modify `prometheusremotewrite.FromMetrics` to not generate `target_info` unless there are metrics, as you can't detect a timestamp otherwise. Currently, if there are no metrics, a `target_info` metric is generated with the Unix epoch for timestamp.

**Link to tracking Issue:** <Issue number if applicable>
See Mimir [issue](https://github.com/grafana/mimir/issues/7272) where this was reported.

**Testing:**
I've added a test case for the bug.

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Review by jesusvazquez — APPROVED** [review]
LGTM

File: .chloggen/bugfix_missing-resource-timestamp.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+subtext:
+
+# If your change doesn't affect end users or the exported elements of any package,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+# Optional: The change log or logs in which this entry should be included.
+# e.g. '[user]' or '[user, api]'
+# Include 'user' if the change is relevant to end users.
+# Include 'api' if there is a change to a library API.
+# Default: '[user]'
+change_logs: [user,api]
```
Comment (dashpole) [inline .chloggen/bugfix_missing-resource-timestamp.yaml:27]: There are no changes to the go library's API here
```suggestion
change_logs: [user]
```

**Review by dashpole — APPROVED** [review]

File: .chloggen/bugfix_missing-resource-timestamp.yaml (hunk 1 above)
Comment (aknuds1) [reply .chloggen/bugfix_missing-resource-timestamp.yaml:27]: Ah yep, I was uncertain as to whether API should be mentioned due to functional changes for API clients. I.e., a client calling `prometheusremotewrite.FromMetrics` with input missing metrics would obtain a different result than before this PR.

**Review by Aneurysm9 — APPROVED** [review]

**Comment by aknuds1** [comment]
CI failure looks like a flake.

**Review by simonpasquier — APPROVED** [review]
Looks like the right approach since there's no way to infer the timestamp anyway.

**Comment by aknuds1** [comment]
CI flaked again.

**Pull request merged**